mesh/mesh.cpp

```diff
@@ -1383,6 +1383,177 @@ void Mesh::GetGeckoElementReordering(Array<int> &ordering,
 #endif
 
 
+struct HilbertCmp
+{
+   int coord;
+   bool dir;
+   const Array<double> &points;
+   double mid;
+
+   HilbertCmp(int coord, bool dir, const Array<double> &points, double mid)
+      : coord(coord), dir(dir), points(points), mid(mid) {}
+
+   bool operator()(int i) const
+   {
+      return (points[3*i + coord] < mid) != dir;
+   }
+};
+
+static void HilbertSort2D(int coord1, bool dir1, bool dir2,
```

Comment:
Nice algorithm here.  Looks like dir1 and dir2 relate to rotating and flipping the quadrants upon recursion, but their meaning isn't that obvious at the outset.  I don't know if there's a comment that would make it obvious or not.

Reply:
Added a couple comments related to this.